tests/workbenches/utils.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright Skodjob authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+from __future__ import annotations
+
+import io
+import pathlib
+from contextlib import contextmanager
+from typing import Generator
+
+from pytest_testconfig import config as py_config
+
+from kubernetes.dynamic import DynamicClient, Resource, ResourceInstance
+
+from ocp_resources.route import Route
+from ocp_resources.notebook import Notebook
+
+from utilities.constants import INTERNAL_IMAGE_REGISTRY_PATH
+
+
+def get_notebook_image(image_name: str, image_tag: str) -> str:
+    controllers_namespace = py_config["applications_namespace"]
+    if py_config.get("distribution") == "upstream":
+        image_dict = {"jupyter-minimal-notebook": "jupyter-minimal-notebook"}
+    else:
+        image_dict = {"jupyter-minimal-notebook": "s2i-minimal-notebook"}
+    return f"{INTERNAL_IMAGE_REGISTRY_PATH}/{controllers_namespace}/{image_dict[image_name]}:{image_tag}"
+
+
+def load_default_notebook(dyn_client: DynamicClient, namespace: str, name: str, image: str) -> Notebook:
+    notebook_string = (pathlib.Path(__file__).parent / "notebook-controller/test_data/notebook.yaml").read_text()
+    notebook_string = notebook_string.replace("my-project", namespace).replace("my-workbench", name)
+    # Set new Route url
+    route_name = "odh-dashboard" if py_config.get("distribution") == "upstream" else "rhods-dashboard"
+    route_host: str = list(
```

Comment:
For better error handeling I think you should do:
```
route = Route(client=dyn_client, name=route_name, namespace=py_config["applications_namespace"])
if not route.exists:
     assert .......
```

with the current code you will get `IndexError: list index out of range` which say nothing.

Reply:
done